src/redux/session/selectors.js

```diff
@@ -39,6 +40,27 @@ export const BALANCES_COMPARATOR_URGENCY = (item1, item2) => {
   return s1 < s2 ? -1 : (s1 > s2 ? 1 : 0)
 }
 
+export const getProfileTokensList = () => createSelector(
+  [ getProfile, getTokens, getMainWallet ],
+  (profile, tokens, mainWallet) => {
+
+    const symbolToTitle = (symbol) => {
+      const titleList = [
+        { symbol: 'XEM', title: 'NEM' },
+      ]
+      const newTitle = titleList.find((s) => s.symbol === symbol)
+      return newTitle?newTitle.title:symbol
+    }
+
+    const addressesInWallet = mainWallet.addresses()
+    return rebuildProfileTokens(profile, tokens)
+      .filter((token) => PROFILE_PANEL_TOKENS.includes(token.symbol))
+      .map((token) => {
+        return { ...token, title: symbolToTitle(token.symbol), address: addressesInWallet.item(token.blockchain).address() }
```

Comment:
Bad idea
Our TokenModel is an immutable model.
Pass here an object with a new title & reference to TokenModel